analysis/tsneLSF.py

```diff
@@ -0,0 +1,334 @@
+    df['tsne-y'] = ty
+    fine_labels = equality.tolist()
+    df['isCorrect'] = fine_labels
+
+    matplotlib.pyplot.figure(figsize=(16,10))
+    sns_plot = sns.scatterplot(
+        x="tsne-x", y="tsne-y",
+        hue='isCorrect',
+        palette=sns.color_palette("hls", len(set(fine_labels))),
+        data=df,
+        legend="full"
+    )
+    fig = sns_plot.get_figure()
+    fig.savefig(os.path.join(path, file_prefix+"_tsne_correctPrediction.png"))
+
+
+def plot_tsne_dots(dataloader, tx, ty, path, file_prefix, legend_labels=[CONSTANTS.DISENTANGLER_CLASS_OUTPUT], phylomapper=None):
+    labels = {}
+    file_names=[]
+    for i, batch in tqdm(enumerate(dataloader), total=len(dataloader)):
+        file_name = batch['file_path_']
+        for j in legend_labels:
+            labels[j] = batch[j] if j not in labels else torch.cat([labels[j], batch[j]]).detach()
+        file_names = file_names + file_name
+
+
+    df = pd.DataFrame()
+    df['tsne-x'] = tx
+    df['tsne-y'] = ty
+    for j in legend_labels:
+        if phylomapper is not None:
+            labels[j] = phylomapper.get_mapped_truth(labels[j])
+            
+        labels[j] = labels[j].tolist()
+        df[j] = labels[j]
+        
+    
+    if phylomapper is not None:
+        file_prefix = file_prefix+"_level"+str(phylomapper.level)
+    
+    for j in legend_labels:    
+        matplotlib.pyplot.figure(figsize=(16,10))
+        sns_plot = sns.scatterplot(
+            x="tsne-x", y="tsne-y",
+            hue=j,
+            palette=sns.color_palette("hls", len(set(labels[j]))),
+            data=df,
+            legend= False#"full"
+        )
+        # tooltip_label = [str(j) for i in range(len(labels))] #TODO: is this i a j?
+        # tooltip = mpld3.plugins.PointLabelTooltip(sns_plot, labels=tooltip_label)
+        fig = sns_plot.get_figure()
+        # mpld3.plugins.connect(fig, tooltip)
+        fig.savefig(os.path.join(path, file_prefix+"_legend_" + j +"_tsne_dots.png"))
+
+
+def visualize_tsne_images(dataloader, tx, ty, img_res, path, file_prefix, phylomapper, cuda=None):    
+    images = None
+    fine_labels=None
+    file_names=[]
+    for i, batch in tqdm(enumerate(dataloader), total=len(dataloader)):
+        image2 = batch['image']
+        fine_label = batch['class']
+        file_name = batch['file_path_']
+        
+            
+        if phylomapper is not None:
+            fine_label = phylomapper.get_mapped_truth(fine_label)
+        
+        if cuda is not None:
+            image2 = image2.cuda()
+        image2 = image2.permute(0, 3, 1, 2).to(memory_format=torch.contiguous_format)
+        image2 = ((image2 + 1.0)*127.5).clip(0,255).type(torch.uint8)
+        images = image2 if images is None else torch.cat([images, image2]).detach()
+        fine_labels = fine_label if fine_labels is None else torch.cat([fine_labels, fine_label]).detach()
+        file_names = file_names + file_name
+
+    # Construct the images
+    width = 12000
```

Comment:
This file is still not cleaned up. Also, please if it is mostly same as tsne.py, try to keep one file and just make necessary changes. If the diff is significant enough, it is fine

Reply:
I've removed the extra comment lines that were left. 
The difference is not significant, only a few changes, but Mridul has also done some changes from his end on this file for cw, so thought once all clean up is done, we could merge it into one file. Is that fine?

Reply:
It is much more efficient to do it now. because we will have more conflicts later

Reply:
Have updated the tsne.py file to be common for all three. Modified loading utils, had to combine the three functions into one. 
A minor change need to be done in the tsne-vqgan.yaml. The `isOriginalVQGAN: true` need to be replaced with `model_name: 'VQGAN'`. Everything else should be fine